cli/azd/pkg/ai/python_bridge.go

```diff
@@ -0,0 +1,143 @@
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/python"
+	"github.com/azure/azure-dev/cli/azd/resources"
+)
+
+// ScriptPath is a type to represent the path of a Python script
+type ScriptPath string
+
+const (
+	// PromptFlowClient is the path to the PromptFlow Client Python script
+	PromptFlowClient ScriptPath = "pf_client.py"
+	// MLClient is the path to the ML Client Python script
+	MLClient ScriptPath = "ml_client.py"
+)
+
+// PythonBridge is an interface to execute python components from the embedded AI resources project
+type PythonBridge interface {
+	Initialize(ctx context.Context) error
+	RequiredExternalTools(ctx context.Context) []tools.ExternalTool
+	Run(ctx context.Context, scriptName ScriptPath, args ...string) (*exec.RunResult, error)
+}
+
+// pythonBridge is a bridge to execute python components from the embedded AI resources project
+type pythonBridge struct {
+	azdCtx        *azdcontext.AzdContext
+	env           *environment.Environment
+	pythonCli     *python.PythonCli
+	commandRunner exec.CommandRunner
+	workingDir    string
+	initialized   bool
+}
+
+// NewPythonBridge creates a new PythonBridge instance
+func NewPythonBridge(
+	azdCtx *azdcontext.AzdContext,
+	env *environment.Environment,
+	pythonCli *python.PythonCli,
+	commandRunner exec.CommandRunner,
+) PythonBridge {
+	return &pythonBridge{
+		azdCtx:        azdCtx,
+		env:           env,
+		pythonCli:     pythonCli,
+		commandRunner: commandRunner,
+	}
+}
+
+// Initialize initializes the PythonBridge
+// Copies embedded AI script files, creates a python virtual environment and installs the required dependencies
+func (b *pythonBridge) Initialize(ctx context.Context) error {
+	if b.initialized {
+		return nil
+	}
+
+	if err := b.initPython(ctx); err != nil {
+		return fmt.Errorf("failed initializing Python: %w", err)
+	}
+
+	b.initialized = true
+
+	return nil
+}
+
+// RequiredExternalTools returns the required external tools for the PythonBridge
+func (b *pythonBridge) RequiredExternalTools(ctx context.Context) []tools.ExternalTool {
+	return []tools.ExternalTool{b.pythonCli}
+}
+
+// Run executes the specified python script with the given arguments
+func (b *pythonBridge) Run(ctx context.Context, scriptName ScriptPath, args ...string) (*exec.RunResult, error) {
+	allArgs := append([]string{string(scriptName)}, args...)
+	runArgs := exec.
+		NewRunArgs("python", allArgs...).
+		WithCwd(b.workingDir)
+
+	result, err := b.commandRunner.Run(ctx, runArgs)
+	if err != nil {
+		return nil, fmt.Errorf("failed running Python script: %w", err)
+	}
+
+	return &result, nil
+}
+
+// initPython initializes the Python environment
+func (b *pythonBridge) initPython(ctx context.Context) error {
+	envDir := b.azdCtx.EnvironmentRoot(b.env.Name())
+	targetDir := filepath.Join(envDir, "ai")
+	if _, err := os.Stat(targetDir); errors.Is(err, os.ErrNotExist) {
```

Comment:
ignoring other `Stat()` errors?

Reply:
Yes, we really just want to check if the folder exists.  If we determine the error to be `ErrNotExist` then we can create the directory.  We shouldn't really expect any permission or other issues here.